coreos-koji-tagger/coreos_koji_tagger.py

```diff
@@ -436,6 +434,20 @@ def process_lockfiles(self, commit=None):
                 watch_tasks(self.koji_client, [task], poll_interval=10)
                 logger.info('Dist-repo task has finished')
 
+    def koji_login(self):
+        # If no keytab then just quietly return
+        if not self.keytab_file:
+            return
+        # If already authenticated then nothing to do
+        if self.koji_client.getLoggedInUser():
+            return
+        # Assert the defined keytab file exists
+        if not os.path.exists(self.keytab_file):
```

Comment:
Hmm, let's check this up front at `__init__` time?

Reply:
we do check it at init time AND here. At init time we check it and print out a useful message to the logs. Here we just silently ignore it. Basically running with no keytab wile is a valid case.

One alternative is that we conditionalize the call to `koji_login()` to only call it if the keytab file var is populated.

```
if self.keytab_file:
    self.koji_login()
```

Reply:
What I mean is checking that the keytab file itself exists. Right now, we're checking this on every re-login instead of once on init. Which is fine, just interesting I guess, since it implies we're guarding ourselves from the keytab being deleted while we're running live, which is "above and beyond the call of duty" for this kind of code. :)

Reply:
But we're not checking if the keytab file exists. We're checking if the `self.keytab_file` variable is not empty (lower cost than checking if the file exists). I don't know of a better way to do the conditional logic.

Reply:
wow.. I totally mistook your original comment.

Reply:
ok addressed now. Sorry for misunderstanding the original comment